utils/python/CIME/case.py

```diff
@@ -6,49 +6,84 @@
-from CIME.XML.env_mach_specific import EnvMachSpecific
-from CIME.XML.env_case          import EnvCase
-from CIME.XML.env_mach_pes      import EnvMachPes
-from CIME.XML.env_build         import EnvBuild
-from CIME.XML.env_run           import EnvRun
-from CIME.XML.env_archive       import EnvArchive
-from CIME.XML.env_batch         import EnvBatch
+from shutil import copyfile
+from CIME.utils                     import expect, run_cmd, get_cime_root, convert_to_type, get_model
+from CIME.XML.machines              import Machines
+from CIME.XML.pes                   import Pes
+from CIME.XML.files                 import Files
+from CIME.XML.component             import Component
+from CIME.XML.compsets              import Compsets
+from CIME.XML.grids                 import Grids
+from CIME.XML.batch                 import Batch
+from CIME.XML.pio                   import PIO
+from CIME.XML.archive               import Archive
+
+from CIME.XML.env_test              import EnvTest
+from CIME.XML.env_mach_specific     import EnvMachSpecific
+from CIME.XML.env_case              import EnvCase
+from CIME.XML.env_mach_pes          import EnvMachPes
+from CIME.XML.env_build             import EnvBuild
+from CIME.XML.env_run               import EnvRun
+from CIME.XML.env_archive           import EnvArchive
+from CIME.XML.env_batch             import EnvBatch
+
+from CIME.XML.generic_xml           import GenericXML
 
 logger = logging.getLogger(__name__)
 
 class Case(object):
 
     def __init__(self, case_root=os.getcwd()):
-        expect(os.path.isdir(case_root),
-               "Case root directory '%s' does not exist" % case_root)
-
-        self._env_entryid_files = []
-        self._env_generic_files = []
 
         self._env_files_that_need_rewrite = set()
 
+        self._env_entryid_files = []
         self._env_entryid_files.append(EnvRun(case_root))
         self._env_entryid_files.append(EnvBuild(case_root))
         self._env_entryid_files.append(EnvMachPes(case_root))
         self._env_entryid_files.append(EnvCase(case_root))
         self._env_entryid_files.append(EnvBatch(case_root))
         if os.path.isfile(os.path.join(case_root,"env_test.xml")):
             self._env_entryid_files.append(EnvTest(case_root))
+        self._env_generic_files = []
         self._env_generic_files.append(EnvMachSpecific(case_root))
         self._env_generic_files.append(EnvArchive(case_root))
+        self._files = self._env_entryid_files + self._env_generic_files
 
-        self._case_root = case_root
+        # Hold arbitary values. In create_newcase we may set values
+        # for xml files that haven't been created yet. We need a place
+        # to store them until we are ready to create the file. At file
+        # creation we get the values for those fields from this lookup
+        # table and then remove the entry. This was what I came up
+        # with in the perl anyway and I think that we still need it here.
+        self.lookups = {}
```

Comment:
lookups is a place to save (and find) values that need to be written to env object that haven't yet been initialized.